dipy/nn/deepn4.py

```diff
@@ -0,0 +1,332 @@
+                                      use_bias=False)
+        self.instnorm = InstanceNormalization(
+            axis=-1, center=False, scale=False)
+        self.activation = LeakyReLU(0.01)
+
+    def call(self, input):
+        x = self.conv3d(input)
+        x = self.instnorm(x)
+        x = self.activation(x)
+
+        return x
+
+
+def UNet3D(input_shape):
+    inputs = tf.keras.Input(input_shape)
+    # Encode
+    x = EncoderBlock(32, kernel_size=3,
+                     strides=1, padding='same')(inputs)
+    syn0 = EncoderBlock(64, kernel_size=3,
+                        strides=1, padding='same')(x)
+
+    x = MaxPool3D()(syn0)
+    x = EncoderBlock(64, kernel_size=3,
+                     strides=1, padding='same')(x)
+    syn1 = EncoderBlock(128, kernel_size=3,
+                        strides=1, padding='same')(x)
+
+    x = MaxPool3D()(syn1)
+    x = EncoderBlock(128, kernel_size=3,
+                     strides=1, padding='same')(x)
+    syn2 = EncoderBlock(256, kernel_size=3,
+                        strides=1, padding='same')(x)
+
+    x = MaxPool3D()(syn2)
+    x = EncoderBlock(256, kernel_size=3,
+                     strides=1, padding='same')(x)
+    x = EncoderBlock(512, kernel_size=3,
+                     strides=1, padding='same')(x)
+
+    # Last layer without relu
+    x = Conv3D(512, kernel_size=1,
+               strides=1, padding='same')(x)
+
+    x = DecoderBlock(512, kernel_size=2,
+                     strides=2, padding='valid')(x)
+
+    x = Concatenate()([x, syn2])
+
+    x = DecoderBlock(256, kernel_size=3,
+                     strides=1, padding='same')(x)
+    x = DecoderBlock(256, kernel_size=3,
+                     strides=1, padding='same')(x)
+    x = DecoderBlock(256, kernel_size=2,
+                     strides=2, padding='valid')(x)
+
+    x = Concatenate()([x, syn1])
+
+    x = DecoderBlock(128, kernel_size=3,
+                     strides=1, padding='same')(x)
+    x = DecoderBlock(128, kernel_size=3,
+                     strides=1, padding='same')(x)
+    x = DecoderBlock(128, kernel_size=2,
+                     strides=2, padding='valid')(x)
+
+    x = Concatenate()([x, syn0])
+
+    x = DecoderBlock(64, kernel_size=3,
+                     strides=1, padding='same')(x)
+    x = DecoderBlock(64, kernel_size=3,
+                     strides=1, padding='same')(x)
+
+    x = DecoderBlock(1, kernel_size=1,
+                     strides=1, padding='valid')(x)
+
+    # Last layer without relu
+    out = Conv3DTranspose(1, kernel_size=1,
+                          strides=1, padding='valid')(x)
+
+    return Model(inputs, out)
+
+
+class DeepN4:
+    """
+    This class is intended for the DeepN4 model.
+    """
+
+    @doctest_skip_parser
+    def __init__(self, verbose=False):
+        r"""
+
+        To obtain the pre-trained model, use::
+        >>> deepn4_model = DeepN4() # skip if not have_tf
+        >>> fetch_model_weights_path = get_fnames('model_weights')
```

Comment:
I think the example should be just using fetch_default_weights function?

Reply:
I changed this to 
 ```
To obtain the pre-trained model, use fetch_default_weights() like:
        >>> deepn4_model = DeepN4()
        >>> fetch_model_weights_path = get_fnames('deepn4_default_weights')
        >>> deepn4_model.load_model_weights(fetch_model_weights_path)
```
let me know if this is good!

Reply:
The example description talks about using fetch_default_weights() but the actual example does not use fetch_default_weights function.